File: qb-engineer-ui/src/app/features/parts/models/update-part-request.model.ts
import { PartStatus } from './part-status.type';
import { PartType } from './part-type.type';

export interface UpdatePartRequest {
  description?: string;
  revision?: string;
  status?: PartStatus;
  partType?: PartType;
  material?: string;
  moldToolRef?: string;
  externalPartNumber?: string;
  toolingAssetId?: number;
  minStockThreshold?: number;
  reorderPoint?: number;
  reorderQuantity?: number;
  leadTimeDays?: number;
  safetyStockDays?: number;
}

File: qb-engineer-server/qb-engineer.core/Models/UpdatePartRequestModel.cs
using QBEngineer.Core.Enums;

namespace QBEngineer.Core.Models;

public record UpdatePartRequestModel(
    string? Description,
    string? Revision,
    PartStatus? Status,
    PartType? PartType,
    string? Material,
    string? MoldToolRef,
    string? ExternalPartNumber,
    int? ToolingAssetId,
    int? PreferredVendorId,
    decimal? MinStockThreshold,
    decimal? ReorderPoint,
    decimal? ReorderQuantity,
    int? LeadTimeDays,
    int? SafetyStockDays);
